### DISCUSSION TITLE
What will happen to my pull requests if I unlink my work email from my work GitHub account?

### DISCUSSION THREAD

**Discussion by WilliamBonvini** [community/community#64114] [Other Feature Feedback, Questions, & Ideas]
Hi,
Unfortunately, I can no longer log in to my work GitHub account (my old phone is broken and I lost my recovery codes). 
It seems that the only solution to my problem would be to create a new account and ask my organization to give me read permissions to our repositories again. 
I need to [unlink my work email](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/unlinking-your-email-address-from-a-locked-account) from my (inaccessible) old account before creating a new one because I'd need to keep using my work email in my new account. 

my question is: **what will happen to the pull requests that are still open (but also to those already merged in master) created by my old account?** My guess is that such pull requests will still be available in our repository and that they simply will be associated with an inaccessible account. 
**It is also my understanding that my old account will continue to exist, but it will simply be inaccessible, correct?**

Thank you in advance for your help!

**Comment by dushmanta05** [comment]
Hi William!
 
Since your account is inaccessible, I guess you've forgotten your password and lost the recovery codes. You can contact GitHub support and show them proof that you're the owner. Maybe they can help you.
 
I'm also curious how you will remove your work email from your inaccessible account. You can't create two accounts with the same email.
 
As far as I know, removing an email removes all the commits associated with that email from the GitHub account, so I'm guessing your pull requests and merged ones will show an unknown account if removing the email works.

**Reply by WilliamBonvini** [reply]
Hi Dushmanta :) 
Correct, I've lost my recovery codes. 

I've thoroughly reviewed the support documentation, and according to GitHub's suggestions, I can either unlink my email to create a new account or temporarily disable 2-factor authentication for my old account. Please note that the latter option may not be available starting from mid-September 2023. Unfortunately, GitHub does not offer chat or phone support for assistance.

If you want to know how to unlink an email from an account I provided the link in my question :)

Regarding your response: 
Are you certain about that? It doesn't seem logical to me that commits would be deleted, as this would disrupt the commit history of our repositories. Many of my commits have already been merged into the master branch.

The commit object contains the following information:

- Directory tree object hash
- Parent commit hash
- Author
- Committer
- Date and message

It would make more sense to me if this information were retained, with only the author and committer fields now pointing to a non-active user.

**Reply by dushmanta05** [reply]
Hi! I actually didn't see the link, but I now understand how to unlink it. Thanks.
 
And about the commits, I said they'll be removed from your account if you unlink that email. This I know because a few months ago I changed my GitHub account email, and once I removed the older email, all the green square contributions were gone from my account, so that's when I thought removing the email would remove commits. But wait, I just realized the green contributions were removed, but I never checked about the commits.
 
Let me do a few tweaks in my repo because that old email is still associated as a secondary email. I'll get back to you after doing some experiments to see whether it removes commits or not.

**Reply by WilliamBonvini** [reply]
Thank you :) correct, I don't care about preserving my repo contributions, I'm interested in the commit history and in the existence of the pull requests (both open and closed). 
Thank you for checking!!

**Reply by dushmanta05** [reply]
Hi again, William!

I've attempted to remove my old email ID and then add it again. I'll share the conclusions and results in this detailed post.

 - Here's how my contributions looked *before removing my email* (174 contributions):
![git.png](https://github.com/community/community/assets/73304098/9dffbd4f-acc7-419a-8bd2-702113a32def)

- *After removing my email*, the contributions looked like this (109 contributions):
![after removing email.png](https://github.com/community/community/assets/73304098/a252e02f-b229-4456-ac91-69aa14d3ac13)

Next, I checked the repositories.

- This is how one of my repositories looked before removing my email (see the portion outlined in red):
![repo-1](https://github.com/community/community/assets/73304098/d6479410-a7e9-48ca-9eab-92da421f06e7)

- After removing the email, the repository appeared like this (I realize this wasn't your question; I'm sharing my observation):
![Screensnhot](https://github.com/community/community/assets/73304098/889c1dd3-d5bc-4c5c-9689-1c2cd738952f)

- I've cropped the image, but the entire commit history and code were present. The only difference, as you can see, is that initially, the commits were attributed to my account (username: dushmanta05). Now, the commits show the name of the author only.

- Here's how the entire commit history looks without the email associated with it for the above repository:
![Screenshot](https://github.com/community/community/assets/73304098/17cc9157-8107-4ab0-9f68-2418b9a08d74)

Now, let's examine the PRs.

#### **Merged PR**

- This is how one of my merged PRs looked before removing the email:
![PR Merged - 1](https://github.com/community/community/assets/73304098/fb9a96d9-cef6-4c13-a5a8-54c6b5b2a542)

- And after removing my email:
![PR Merged - 2](https://github.com/community/community/assets/73304098/e118a084-f3b0-4018-866f-243fabb05a76)

- Do you see the difference in the above two pictures? There's no apparent difference, so I believe PRs remain even after removing the email.

- However, in the PR, the commits now only show the author name, as mentioned in the previous point. For confirmation, refer to the images below.

- Before removing email: 
![Commit - 1](https://github.com/community/community/assets/73304098/68b4afcf-2d38-4314-9baa-2d15aff25553)

- After removing the email:
![Commit -2](https://github.com/community/community/assets/73304098/fc014b3d-9b50-4ec6-ad8f-0f3a04595108)

#### **Open PR & Closed PR**

- Similar to merged PRs, here the requests remain associated with your name, but the commit authorship changes to the actual author's name. I have screenshots, but I'm not including them here.

- To double-verify, I checked if the merged PRs were present in the main repository, and I can confirm that they are present under my username.

- Here's a screenshot confirming that the main repository contains the PR with my username:
![Screenshot](https://github.com/community/community/assets/73304098/51b4986f-c6ce-4700-9b00-6dee2fca9a06)

After adding my email address back again, all the contributions were shown on my profile. I believe that if you were to create a new account with your work email, all the contributions would show up on the new account as well. 

Also, I forgot to mention that all the issues I created were also present in my account even after removing my email.

Hope it helps :)

**Reply by WilliamBonvini** [reply]
Hi Dushmanta, thank you for the very clear and complete response!
I just have a follow-up question: what do you mean by "I've attempted to remove my old email ID?" Do you mean that you unlinked it from your current account? 
I will mark your answer as the accepted one once we untie this last knot, because, from what I understand, it seems that unlinking an email from an account has no repercussions on commit history and pull requests

**Reply by dushmanta05** [reply]
Yes, I meant unlinking when I said removing the email ID. I have two email addresses linked: one is the primary address that I recently added, and I've been using it for all my commits. I unlinked my secondary email, which I used for all the commits a few months ago.

![Screenshot](https://github.com/community/community/assets/73304098/580e5f4a-e6e3-4821-9670-1b69d5fc9d77)

And thank you for marking it as the answer :)

**Reply by WilliamBonvini** [reply]
Perfect, thank you, everything's clear ;)

**Reply by dushmanta05** [reply]
You're welcome.

**Reply by mrthawzinoo** [reply]
How can i recover unlink primary account and secondary account?

**Reply by dushmanta05** [reply]
@mrthawzinoo Could you please explain more? Thanks.

**Reply by mrthawzinoo** [reply]
Gitthub advised me to give myself two days to heal after I received a two factor backup lock. I unlinked the email because I could not wait 3days. My account is not connected to the email at that point. I can no longer enter the email I use to sign in to github. I reached out to github, but I am unable to utilize its community support. How can I get my account a return? I need Github assistance.

**Reply by dushmanta05** [reply]
@mrthawzinoo As I understand it, you unlinked your email from your GitHub account and are now unable to add it again, correct? I would recommend checking to see whether the email is linked to any other accounts. If it isn't connected, I'm not sure why it's not happening. You could reach out to GitHub support. Thanks

**Reply by mrthawzinoo** [reply]
you unlinked your email from your GitHub account and are now unable to add it again, correct?
Yes , sure . 

But there is no email connection and I can't access it anymore. If I'm using github free version, I can't talk to them github support directly. I'm really upset now.
Thank for your reply .

**Reply by dushmanta05** [reply]
@mrthawzinoo if you've got the username, then you can login with username and password, I think. And I didn't know it was possible to remove the primary address from an email.

**Reply by dushmanta05** [reply]
@mrthawzinoo I just checked, and it's not possible to remove it for me. You've got to keep at least one email in your GitHub account; you can't remove the only primary email address.

**Reply by mrthawzinoo** [reply]
Yes, when I unlinked it, I can no longer use either the primary email  and the secondary email. I can only signin with one username. But my account is auth locked, so I need to make a backup.
<img width="1152" alt="Screenshot 2024-05-12 at 23 23 04" src="https://github.com/community/community/assets/169356810/1e16480e-93c1-4dc6-b818-c9d82b0f6d61">

**Reply by dushmanta05** [reply]
@mrthawzinoo Yeah! I get it now. I think you should create a support ticket with the new account (this one, I believe) and see if it works.
